extensions/data-workspace/src/api/git.d.ts

```diff
@@ -0,0 +1,312 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
to use the git extension apis, the git.d.ts needs to be copied into the extension as specified in the git extension's [README](https://github.com/microsoft/vscode/tree/main/extensions/git)

Reply:
That's only for extensions that don't share the repo - you can add a direct reference to the typings file in the ref.d.ts such as we do in other places such as https://github.com/microsoft/azuredatastudio/blob/main/extensions/sql-migration/src/typings/ref.d.ts#L10

e.g. for this it'd be 

`/// <reference path='../../../git/src/api/git.d.ts'/>`

Reply:
there isn't a module declared in git.d.ts, so there was a "Cannot find module 'git' or its corresponding type declarations." error when I tried doing that. Is there another way to get this to work besides changing the git.d.ts to declare a module(I looked into that before, but didn't want to make too many changes in the git extension for merge conflicts int he future)? The github extension in the vscode repo also uses the git apis by having a copy of git.d.ts.

Reply:
where is the error being raised?

Reply:
it's a compilation error

Reply:
I opened a PR https://github.com/microsoft/vscode/pull/121979 in the vscode repo to wrap the git apis in a module, but I didn't want to hold up these changes to wait or hope for that to get in. I can also make these changes in ADS, but I wasn't sure if that would be worth the merge conflicts in the future.

Reply:
Gotcha - yeah that's unfortunate. But you can still reference the typings file directly in the import.

```
import * as git from '../../git/src/api/git';
...
const gitApi = (<git.GitExtension>vscode.extensions.getExtension('')?.exports).getAPI(1);
```

The reason we should avoid duplicating the typings file is because it'd be really annoying to keep it in sync with the actual typings file in the git extension during VS Code merges. So if something changed there and we missed updating the one you added then it could break your extension.

Reply:
another option is to add a precompile step to copy the typings file, we will get a compilation error if the git typings file is no longer working with our code. this is better than a runtime error.

Reply:
That's just unnecessary overhead for something that lives in this repo though *shrug*